### ISSUE TITLE
TCK Challenge : Some tests in EE8 are depended on JavaSE 8 classes and/or not portable to higher JDKs

### ISSUE THREAD

**Issue by alwin-joseph** [jakartaee/platform-tck#1190]
**Follow the TCK Process document**
Please follow the instructions in the TCK Process document (https://jakarta.ee/committees/specification/tckprocess) which may be updated occasionally.

**Filing a Challenge**

TCK Challenges MUST be filed via the specification project’s issue tracker using the label challenge and include the following information:

1. The relevant specification version and section number(s) : 

https://jakarta.ee/specifications/platform/8

2. The coordinates of the challenged test(s) : 

- [src/com/sun/ts/tests/jsonb/api/annotation/AnnotationTest.java#testJsonbDateFormat](https://github.com/jakartaee/platform-tck/blob/c457c40af61d307b0dea0e0489fed7ab4c3e8490/src/com/sun/ts/tests/jsonb/api/annotation/AnnotationTest.java#L75C23-L75C23)

- https://github.com/jakartaee/platform-tck/blob/8.0.2/src/com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java

      com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java#signatureTest_from_appclient
      com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java#signatureTest_from_ejb
      com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java#signatureTest_from_jsp
      com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java#signatureTest_from_servlet
      com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java#signatureTest_from_standalone

-  [src/com/sun/ts/tests/jaxrpc/api/javax_xml_rpc/Service/Client.java#GetPortConstructor2NegTest1WithWsdl](https://github.com/jakartaee/platform-tck/blob/c457c40af61d307b0dea0e0489fed7ab4c3e8490/src/com/sun/ts/tests/jaxrpc/api/javax_xml_rpc/Service/Client.java#L1161)

-  [src/com/sun/ts/tests/jaxws/api/javax_xml_ws/Service/Client.java#GetPort2NegTest1WithWsdl](https://github.com/jakartaee/platform-tck/blob/c457c40af61d307b0dea0e0489fed7ab4c3e8490/src/com/sun/ts/tests/jaxws/api/javax_xml_ws/Service/Client.java#L846)

-  [src/com/sun/ts/tests/webservices/deploy/GenSvc/Client.java#GetPortConstructor2NegTest1](https://github.com/jakartaee/platform-tck/blob/c457c40af61d307b0dea0e0489fed7ab4c3e8490/src/com/sun/ts/tests/webservices/deploy/GenSvc/Client.java#L755)

-  [src/com/sun/ts/tests/webservices/deploy/ServiceNPA/Client.java#GetPortConstructor2NegTest1](https://github.com/jakartaee/platform-tck/blob/c457c40af61d307b0dea0e0489fed7ab4c3e8490/src/com/sun/ts/tests/webservices/deploy/ServiceNPA/Client.java#L705)

- [src/com/sun/ts/tests/webservices/deploy/ServicePW/Client.java#GetPortConstructor2NegTest1](https://github.com/jakartaee/platform-tck/blob/c457c40af61d307b0dea0e0489fed7ab4c3e8490/src/com/sun/ts/tests/webservices/deploy/ServicePW/Client.java#L518)

5. The exact TCK version 

https://download.eclipse.org/jakartaee/platform/8/jakarta-jakartaeetck-8.0.2.zip 

8. The implementation being tested, including name and company.

Weblogic Server 14.1.2 
Company : Oracle

9. A full description of why the test is invalid and what the correct behavior is believed to be.

- The webservices/jaxrpc/jaxws tests mentioned above are depended on Java SE 8 class java.rmi.activation.Activator which was removed in JDK17. The commit https://github.com/jakartaee/platform-tck/commit/cc74750f95ae0d5591dbc9e50a3a12351dabd64b was made in EE10 to remove this dependency.
- The jsonb test has fix made in EE9 using https://github.com/jakartaee/platform-tck/pull/152 to work on JDK11+.
- The signature tests are depended on SE8 classes in EE8, a fix for the same has been made in https://github.com/jakartaee/platform-tck/pull/720 , issue https://github.com/jakartaee/platform-tck/issues/156 in EE9.

10. Any supporting material; debug logs, test output (.jtr file), test logs, run scripts, etc.

Will provide test run logs or output for these tests very soon. 

**Additional context**

Suggesting to provide solutions to all the above test failures in JakartaEE8 TCK (and JAXWS TCK) by back porting them from EE9/9.1/10.0 TCK so they can work on other Java SE versions as well. The fix will not affect the implementations which already passed and certified using the EE8 TCK. This will enable more Compatible implementations to certify for JakartaEE8 which currently runs on higher JDKs.

**Comment by scottmarlow** [comment]
The Platform team has documented that we don't support more than the past two releases.  Still, IMO there is value in what you are proposing in this challenge for continuing to validate that EE 8 implementations are compatible on higher JDKs.

Note that addressing this challenge will involve adding any needed CI jobs to the https://ci.eclipse.org/jakartaee-tck/job/8/ folder which is currently empty.

**Comment by starksm64** [comment]
I would say just exclude them as the minimal effort solution.

**Comment by alwin-joseph** [comment]
> The Platform team has documented that we don't support more than the past two releases. Still, IMO there is value in what you are proposing in this challenge for continuing to validate that EE 8 implementations are compatible on higher JDKs.
> 
> Note that addressing this challenge will involve adding any needed CI jobs to the https://ci.eclipse.org/jakartaee-tck/job/8/ folder which is currently empty.

Agree, jenkins jobs would be required to generate the next TCK version. Also it will be required to create a new branch based out of 8.0.2 tag so the PRs can be created against them to address the challenge. If the challenge is accepted I should be able to do them both and raise PRs also. 

> I would say just exclude them as the minimal effort solution.

Exclude does sound like easier solution; but considering that the challenged tests include signature tests also which are sort of vital tests, IMHO it would be great if we could backport the signature test tool improvement that was done in TCK 9.0(by Scott M) to not depend on the JDK classes.

**Comment by gurunrao** [comment]
Jenkins Jobs for Platform TCK 8.0.X can be found at https://ci.eclipse.org/jakartaee-tck/job/jakartaee-8.0.x-release/, we can clone the job as required. 
The failing Signature, JAXWS, Webservices, JAXRPC, JSONB tests can be easily back-ported without much effort from 10.X and 9.X branch. Also, to test the generated Platform TCK bundle,  Glassfish 5.1 and JDK8 can be used.

**Comment by gurunrao** [comment]
@alwin-joseph has made progress with back-port of code changes required for Signature, JAXWS, Webservices, JAXRPC, JSONB tests to pass with JDK11+.@alwin-joseph will raise enhancement PR for 8.0.X branch based on consensus within the TCK community.
